src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,48 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder outputString = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo);
+
+        for (String dataEmployeeName : names) {
+            int totalEmployeeSalary = 0;
+            for (String employeeRecord : data) {
+                String[] splittedEmployeeRecord = employeeRecord.split(" ");
+                String recordEmployeeName = splittedEmployeeRecord[1];
+                if (!dataEmployeeName.equals(recordEmployeeName)) {
+                    continue;
+                }
+
+                String recordDate = splittedEmployeeRecord[0];
+                String employeeHoursOfWork = splittedEmployeeRecord[2];
+                String employeeSalaryPerHour = splittedEmployeeRecord[3];
+
+                String[] splittedDateFrom = dateFrom.split("\\.");
+                String[] splittedDateTo = dateTo.split("\\.");
+                String[] splittedRecordDate = recordDate.split("\\.");
+                if ((Integer.parseInt(splittedRecordDate[2])
+                                <= Integer.parseInt(splittedDateFrom[2])
+                            && (Integer.parseInt(splittedRecordDate[1])
+                                <= Integer.parseInt(splittedDateFrom[1])
+                            && (Integer.parseInt(splittedRecordDate[0])
+                                <= Integer.parseInt(splittedDateFrom[0]))))
+                        || (Integer.parseInt(splittedRecordDate[2])
+                                >= Integer.parseInt(splittedDateTo[2])
+                            && (Integer.parseInt(splittedRecordDate[1])
+                                >= Integer.parseInt(splittedDateTo[1])
+                            && (Integer.parseInt(splittedRecordDate[0])
+                                > Integer.parseInt(splittedDateTo[0]))))) {
```

Comment:
The date comparison logic is incorrect. The current logic checks if the record date is less than or equal to the start date and greater than or equal to the end date, which is logically flawed. You should check if the record date is between the start date and end date inclusively. Consider using a more straightforward date comparison approach or a date library to handle this.